kafka-connect/src/main/java/io/tabular/iceberg/connect/data/IcebergWriterFactory.java

```diff
@@ -34,26 +39,107 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+// TODO get the DLT options for partitioning from the config and look into how that works
+
 public class IcebergWriterFactory {
 
   private static final Logger LOG = LoggerFactory.getLogger(IcebergWriterFactory.class);
 
   private final Catalog catalog;
   private final IcebergSinkConfig config;
 
+  private final Predicate<String> shouldAutoCreate;
+
+  private TableIdentifierSupplier tableIdSupplier;
+
+  private TableCreator tableCreator;
+
+  private static class TableIdentifierSupplier {
+    TableIdentifier id(String name) {
+      return TableIdentifier.parse(name);
+    }
+  }
+
+  private static class ErrorHandlingTableIdentifierSupplier extends TableIdentifierSupplier {
+    @Override
+    TableIdentifier id(String name) {
+      TableIdentifier tableId;
+      try {
+        tableId = super.id(name);
+      } catch (Exception error) {
+        throw new DeadLetterUtils.DeadLetterException("TABLE_IDENTIFIER", error);
+      }
+      return tableId;
+    }
+  }
+
+  private static class TableCreator {
+
+    private final Catalog catalog;
+
+    TableCreator(Catalog catalog) {
+      this.catalog = catalog;
+    }
+
+    public Table createTable(
+        TableIdentifier identifier,
+        Schema schema,
+        PartitionSpec spec,
+        Map<String, String> properties) {
+      return catalog.createTable(identifier, schema, spec, properties);
+    }
+  }
+
+  private class ErrorHandlingTableCreator extends TableCreator {
+
+    ErrorHandlingTableCreator(Catalog catalog) {
+      super(catalog);
+    }
+
+    @Override
+    public Table createTable(
+        TableIdentifier identifier,
+        Schema schema,
+        PartitionSpec spec,
+        Map<String, String> properties) {
+      Table table;
+      try {
+        table = catalog.createTable(identifier, schema, spec, properties);
+      } catch (IllegalArgumentException | ValidationException error) {
+        throw new DeadLetterUtils.DeadLetterException("CREATE_TABLE", error);
+      }
+      return table;
+    }
+  }
+
   public IcebergWriterFactory(Catalog catalog, IcebergSinkConfig config) {
     this.catalog = catalog;
     this.config = config;
+    this.tableCreator = new TableCreator(catalog);
+    this.tableIdSupplier = new TableIdentifierSupplier();
+
+    if (config.autoCreateEnabled()) {
+      shouldAutoCreate = (unused) -> true;
+    } else if (config.deadLetterTableEnabled()) {
```

Comment:
Bug / missing functionality here [WIP] 

- this will partition the table as per the data tables, likely fail, and end up with unpartitioned tables (since partitionSpec failures default to unpartitioned) 
- need to add a config option for the partition spec for the Dead Letter Table 
- need to update / add functionality to parse this partition spec (errors should take down the pipeline) and partition the Dead Letter Table on creation like that.

Reply:
This is fine.  We can use 

```
iceberg.table.<table name>.partition-by
```
 
To control the partitioning of the table, no config changes required.